### PR TITLE
python3Packages.python-magic: 0.4.18 -> 0.4.20

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.4.20

- Tests are now enabled
- Follow PEP 0503

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/python-magic/default.nix [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
 
-  meta = {
-    description = "A python interface to the libmagic file type identification library";
+  pythonImportsCheck = [ "magic" ];
+
+  meta = with lib; {
+    description = "Python interface to the libmagic file type identification library";
     homepage = "https://github.com/ahupp/python-magic";
-    license = lib.licenses.mit;
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-magic/default.nix:40]: Maybe you want to add yourself as a maintainer.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -6331,7 +6331,9 @@ in {
 
   python-lzo = callPackage ../development/python-modules/python-lzo { inherit (pkgs) lzo; };
 
-  python_magic = callPackage ../development/python-modules/python-magic { };
+  python-magic = callPackage ../development/python-modules/python-magic { };
+  python_magic = self.python-magic; # Old attr, 2021-02-10
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:6335]: Is this used anywhere in nixpkgs? If not we might change it without an alias.

Comment (fabaff) [reply pkgs/top-level/python-packages.nix:6335]: Is used by those packages:

```
pkgs/applications/networking/irc/weechat/scripts/weechat-matrix/default.nix
pkgs/development/python-modules/alot/default.nix
pkgs/development/python-modules/androguard/default.nix
pkgs/development/python-modules/beancount/default.nix
pkgs/development/python-modules/django-versatileimagefield/default.nix
pkgs/development/python-modules/mocket/default.nix
pkgs/development/python-modules/pushbullet/default.nix
pkgs/development/python-modules/pysnow/default.nix
pkgs/development/python-modules/relatorio/default.nix
pkgs/development/python-modules/thumbor/default.nix
pkgs/servers/mautrix-telegram/default.nix
pkgs/servers/pinnwand/steck.nix
pkgs/tools/misc/diffoscope/default.nix
pkgs/tools/networking/s3cmd/default.nix
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112786` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages marked as broken and skipped:
</summary>
<ul>
<li>ledger2beancount</li>
<li>python39Packages.geoip2</li>
<li>python39Packages.pysnow</li>
<li>python39Packages.sopel</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>diffoscope: log was empty</li>
  </ul>
</details>
<details>

  <summary>39 packages failed to build and are new build failures:</summary>
  <ul>
<li>beancount: log was empty</li>
<li>(python38Packages.beancount): log was empty</li>
<li>fava: log was empty</li>
<li>python37Packages.beancount: log was empty</li>
<li>python37Packages.beancount_docverif: log was empty</li>
<li>python37Packages.django-versatileimagefield: log was empty</li>
<li>python37Packages.geoip2: log was empty</li>
<li>python37Packages.mocket: log was empty</li>
<li>python37Packages.pushbullet: log was empty</li>
<li>python37Packages.pysnow: log was empty</li>
<li>python37Packages.python-magic: log <a href="https://termbin.com/tt8t">https://termbin.com/tt8t</a></li>
<li>(python37Packages.python_magic): log was empty</li>
<li>python37Packages.relatorio: log was empty</li>
<li>python37Packages.sopel: log was empty</li>
<li>python37Packages.trytond: log was empty</li>
<li>python38Packages.beancount_docverif: log was empty</li>
<li>python38Packages.django-versatileimagefield: log was empty</li>
<li>python38Packages.geoip2: log was empty</li>
<li>python38Packages.mocket: log was empty</li>
<li>python38Packages.pushbullet: log was empty</li>
<li>python38Packages.pysnow: log was empty</li>
<li>python38Packages.python-magic: log <a href="https://termbin.com/w607">https://termbin.com/w607</a></li>
<li>(python38Packages.python_magic): log was empty</li>
<li>python38Packages.relatorio: log was empty</li>
<li>python38Packages.sopel: log was empty</li>
<li>trytond: log was empty</li>
<li>(python38Packages.trytond): log was empty</li>
<li>python39Packages.beancount: log was empty</li>
<li>python39Packages.beancount_docverif: log was empty</li>
<li>python39Packages.django-versatileimagefield: log was empty</li>
<li>python39Packages.mocket: log was empty</li>
<li>python39Packages.pushbullet: log was empty</li>
<li>python39Packages.python-magic: log <a href="https://termbin.com/dsmke">https://termbin.com/dsmke</a></li>
<li>(python39Packages.python_magic): log was empty</li>
<li>python39Packages.relatorio: log was empty</li>
<li>python39Packages.trytond: log was empty</li>
<li>s3cmd: log was empty</li>
<li>steck: log was empty</li>
<li>weechatScripts.weechat-matrix: log was empty</li>
  </ul>
</details>

File: pkgs/development/python-modules/python-magic/default.nix [hunk 3]
```diff
@@ -1,28 +1,43 @@
...
-    inherit pname version;
-    sha256 = "b757db2a5289ea3f1ced9e60f072965243ea43a2221430048fd8cacab17be0ce";
+  src = fetchFromGitHub {
+    owner = "ahupp";
+    repo = pname;
+    rev = version;
+    sha256 = "1zk9jqzfnkgmizislyb8nq6vvikylaybf1rrgd25pn8d4sf5iqp0";
   };
 
+  propagatedBuildInputs = [ file ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-magic/default.nix:21]: Why?

File: pkgs/development/python-modules/python-magic/default.nix [hunk 4]
```diff
@@ -1,28 +1,43 @@
...
+
   postPatch = ''
-    substituteInPlace magic.py --replace "ctypes.util.find_library('magic')" "'${file}/lib/libmagic${stdenv.hostPlatform.extensions.sharedLibrary}'"
+    substituteInPlace magic/__init__.py --replace "ctypes.util.find_library('magic')" "'${file}/lib/libmagic${stdenv.hostPlatform.extensions.sharedLibrary}'"
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-magic/default.nix:24]: I prefer using `patches` as that breaks more obviously.

File: pkgs/development/python-modules/python-magic/default.nix [hunk 5]
```diff
@@ -1,28 +1,43 @@
...
   '';
 
-  doCheck = false;
+  checkInputs = [ pytestCheckHook glibcLocales ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-magic/default.nix:27]:
```suggestion
  checkInputs = [ glibcLocales ];
```
Upstream tests don't depend on pytest.

Comment (dotlambda) [inline pkgs/development/python-modules/python-magic/default.nix:27]:
```suggestion
  checkInputs = [ glibcLocales ];
```
Upstream tests don't depend on pytest.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
ping @fabaff

**Comment by dotlambda** [comment]
@fabaff Do you want someone else to take over this pull request?

**Pull request closed without merging**